ee/src/main/java/org/jboss/as/ee/component/ComponentConfiguration.java

```diff
@@ -505,7 +505,7 @@ public void setInstanceFactory(final ComponentFactory instanceFactory) {
         this.instanceFactory = instanceFactory;
     }
 
-    public ClassIndex getClassIndex() {
+    public ClassReflectionIndex getClassIndex() {
```

Comment:
Really ClassReflectionIndex should not be parameterised. I will remove all the <?> from full, so I can make the change in core.